### PR TITLE
Add C implementation for `@stdlib/math/base/special/sinh`

### LINKED ISSUE #747: [RFC]: Add C implementation for `@stdlib/math/base/special/sinh`
### Description

This RFC proposes adding C implementation for [`@stdlib/math/base/special/sinh`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/@stdlib/math/base/special/sinh).

```c
double stdlib_base_sinh( const double x );
```

### Related Issues

None.

### Questions

No.

### Other

No.

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by Pranavchiku**
Resolves #747.

## Checklist

- [x] update readme.md
- [x] include.gypi
- [x] binding.gyp
- [x] include/stdlib/math/base/special/
- [x] evalrational.js
- [x] src
- [x] manifest.json
- [x] lib
- [x] examples
- [x] benchmark
- [x] test
* * *

@kgryte

### CODE REVIEW

File: lib/node_modules/@stdlib/math/base/special/sinh/src/main.c [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+
+#include "stdlib/math/base/special/sinh.h"
+#include "stdlib/math/base/special/abs.h"
+#include "stdlib/math/base/special/exp.h"
+#include "stdlib/constants/float64/pinf.h"
+#include "stdlib/constants/float64/ninf.h"
+#include "stdlib/constants/float64/ln_two.h"
+#include <stdint.h>
+
+static const double POS_OVERFLOW = 7.09782712893383996843e2 + STDLIB_CONSTANT_FLOAT64_LN2;
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/sinh/src/main.c:41]: Here, it was not allowing me to declare like Javascript variables 
```c
static const double MAXLOG = 7.09782712893383996843e2;

// ln(2^-1022)
static const double MINLOG = -7.08396418532264106224e2;

static const double POS_OVERFLOW = MAXLOG + STDLIB_CONSTANT_FLOAT64_LN2;
static const double NEG_OVERFLOW = MINLOG - STDLIB_CONSTANT_FLOAT64_LN2;
```
It was throwing an error at `POS_OVERFLOW` declaration, hence I used these magic numbers.

**Comment by Pranavchiku** [comment]
All tests passed and benchmarks cleared, this PR can have a final review.

File: lib/node_modules/@stdlib/math/base/special/sinh/src/main.c (hunk 1 above)
Comment (kgryte) [reply lib/node_modules/@stdlib/math/base/special/sinh/src/main.c:41]: That should be fine so long as the values are properly documented.

File: lib/node_modules/@stdlib/math/base/special/sinh/README.md [hunk 2]
```diff
@@ -77,6 +77,95 @@ for ( i = 0; i < x.length; i++ ) {
...
+
+<!-- C API usage examples. -->
+
+<section class="examples">
+
+### Examples
+
+```c
+#include "stdlib/math/base/special/sinh.h"
+#include <stdlib.h>
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/sinh/README.md]:
```suggestion
```

The `stdlib.h` header is not used. For a list of what's in `stdlib.h`, see [here](https://en.wikibooks.org/wiki/C_Programming/stdlib.h). In most cases, we'd use `stdlib.h` when generating random numbers.

File: lib/node_modules/@stdlib/math/base/special/sinh/examples/c/example.c [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#include "stdlib/math/base/special/sinh.h"
+#include <stdlib.h>
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/sinh/examples/c/example.c]:
```suggestion
```

**Review by kgryte — APPROVED** [review]
LGTM. Thanks, @Pranavchiku!

**Pull request merged**